radix-engine/tests/metering.rs

```diff
@@ -0,0 +1,40 @@
+#[rustfmt::skip]
+pub mod test_runner;
+
+use radix_engine::ledger::*;
+use radix_engine::transaction::*;
+use radix_engine::wasm::InvokeError;
+use scrypto::call_data;
+use scrypto::prelude::*;
+
+#[test]
+fn test_loop() {
+    let mut ledger = InMemorySubstateStore::with_bootstrap();
+    let mut executor = TransactionExecutor::new(&mut ledger, true);
+    let package = executor
+        .publish_package(&compile_package!(format!("./tests/{}", "metering")))
+        .unwrap();
+
+    let transaction = TransactionBuilder::new()
+        .call_function(package, "Metering", call_data!(iterations(10_000u32)))
+        .build(executor.get_nonce([]))
+        .sign([]);
+    let receipt = executor.validate_and_execute(&transaction).unwrap();
+    assert!(receipt.result.is_ok());
```

Comment:
Can we use `.expect` rather than `assert` as `.expect` displays the actual error on failure

Reply:
Yes yes